src/vocab.rs

```diff
@@ -4,9 +4,11 @@ use std::rc::Rc;
 use finalfusion::chunks::vocab::{NGramIndices, SubwordIndices, VocabWrap, WordIndex};
 use finalfusion::prelude::*;
 use pyo3::class::sequence::PySequenceProtocol;
-use pyo3::exceptions;
 use pyo3::prelude::*;
+use pyo3::types::PyAny;
+use pyo3::{exceptions, PyIterProtocol, PyMappingProtocol};
```

Comment:
I have a small preference for different lines for different submodules, so here:

```rust
use pyo3::exceptions;
use pyo3::{PyIterProtocol, PyMappingProtocol};
```
otherwise you have to be fully consistent:

```rust
use pyo3::{exceptions, types::PyAny, PyIterProtocol, PyMappingProtocol};
```

etc. Which is ugly ;).

Also, keeping per module imports makes diffs smaller.